addons/product/tests/test_product_attribute_value_config.py

```diff
@@ -697,3 +698,27 @@ def test_inactive_related_product_update(self):
         product.action_unarchive()
         self.assertTrue(product.active, 'The product should be unarchived.')
         self.assertEqual(product_attribut.number_related_products, 0, 'The product attribute must not have an associated product')
+
+    def test_copy_extra_prices_of_product_attribute_values(self):
+        """
+            Ensure that exclusions and price extra are copied along the duplication of a product.
+        """
+        product_template = self.computer
+        exclusion = self.env['product.template.attribute.exclusion'].create({
+            'product_tmpl_id': product_template.id,
+            'value_ids': [
+                Command.link(product_template.attribute_line_ids.product_template_value_ids.ids[0])
+            ],
+            })
+        product_template.attribute_line_ids[1].product_template_value_ids[0].exclude_for = [
+            Command.link(exclusion.id)
+        ]
+        extra_prices = product_template.attribute_line_ids.product_template_value_ids.mapped(
+            'price_extra'
+        )
+        copied_template = product_template.copy()
+        copied_extra_prices = copied_template.attribute_line_ids.product_template_value_ids.mapped(
+            'price_extra'
+        )
+        self.assertEqual(extra_prices, copied_extra_prices)
+        self.assertTrue(product_template.attribute_line_ids.product_template_value_ids.exclude_for)
```

Comment:
Don't just check if there is a value here, please ensure it's the expected one =]
It'd be better to have both an exclusion for another product and an exclusion for a ptav of the same template, so that it'd check the 2 different case =]

Reply:
Good suggestion, test updated 😊